torlakian_data.py

```diff
@@ -0,0 +1,258 @@
+            print("Download finished.")
+
+        return path
+
+    def unzip(self, path):
+        """Unzipping of the corpus .zip archive"""
+
+        print("Unzipping...")
+        with zipfile.ZipFile(path, 'r') as zip_ref:
+            directory = os.path.dirname(path)
+            zip_ref.extractall(directory)
+
+        for file in os.listdir(directory):
+            filename = os.fsdecode(file)
+            if filename.endswith(".zip") and filename != os.path.basename(path):
+                full_path = os.path.join(directory, filename)
+                with zipfile.ZipFile(full_path, 'r') as zip_ref:
+                    zip_ref.extractall(directory)
+
+                os.remove(full_path)
+        
+        print("Unzipping finished.")
+    
+    def join_data(self):
+        """Joins all files in the corpus into one big dataset, 
+        then performs 70:20:10 train-val-test split and 
+        saves these into separate files"""
+
+        directory = 'data/TOR_C_TXT_transcripts'
+        all_data = []
+
+        for file in os.listdir(directory):
+            if not file.startswith("TIM_"):
+                continue
+
+            filepath = os.path.join(directory, os.fsdecode(file))
+            if not filepath.endswith(".txt"):
+                continue
+
+            with open(filepath, 'r') as f:
+                lines = f.readlines()
+                lines = lines[1:]
+                lines = [' '.join(l.split()[1:]) for l in lines]
+                all_data.extend(lines)
+
+        num_lines = len(all_data)
+        train = all_data[:int(num_lines * 0.7)]
+        val = all_data[int(num_lines * 0.7): int(num_lines * 0.9)]
+        test = all_data[int(num_lines * 0.9):]
+
+        with open('data/train.txt', 'w') as f:
+            f.write('\n'.join(train))
+        with open('data/val.txt', 'w') as f:
+            f.write('\n'.join(val))
+        with open('data/test.txt', 'w') as f:
+            f.write('\n'.join(test))
+
+    def __init__(self, window: int, alphabet_size: int = 0, clean: bool = False):
+        path = self.download()
+        self.unzip(path)
+        self.join_data()
+
+        for dataset in ["train", "val", "test"]:
+            with open(f"data/{dataset}.txt", "r") as dataset_file:
+                data = dataset_file.read()
+            setattr(self, dataset, self.Dataset(
+                data,
+                window,
+                alphabet=alphabet_size if dataset == "train" else self.train.alphabet,
+                clean=clean
+            ))
+
+    train: Dataset
+    dev: Dataset
+    test: Dataset
+
+    # Evaluation infrastructure.
+    @staticmethod
+    def evaluate(gold_dataset: Dataset, predictions: str) -> float:
+        gold = gold_dataset.text
+
+        if len(predictions) < len(gold):
+            raise RuntimeError("The predictions are shorter than gold data: {} vs {}.".format(
+                len(predictions), len(gold)))
```

Comment:
it is better to use f-strings:
```suggestion
            raise RuntimeError(f"The predictions are shorter than gold data: {len(predictions)} vs {len(gold)}.")
```

Reply:
fixed